File: Hot 100/64. 最小路径和.go
func minPathSum(grid [][]int) int {
	dp := make([][]int, 205)
	for i := range dp {
		dp[i] = make([]int, 205)
	}
	m, n := len(grid), len(grid[0])
	dp[0][0] = grid[0][0]
	for i := 1; i < m; i++ {
		dp[i][0] = dp[i-1][0] + grid[i][0]
	}
	for j := 1; j < n; j++ {
		dp[0][j] = dp[0][j-1] + grid[0][j]
	}
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			dp[i][j] = grid[i][j] + min(dp[i-1][j], dp[i][j-1])
		}
	}
	return dp[m-1][n-1]
}
func min(x, y int) int {
	if x > y {
		return y
	}
	return x
}

File: 刷题/CodeTop/64. 最小路径和.cpp
class Solution {
public:
    int minPathSum(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        vector<vector<int> > dp(m, vector<int>(n, 0));
        for(int i = 0; i < m; i++) {
            if(i == 0) dp[i][0] = grid[i][0];
            else dp[i][0] = dp[i-1][0] + grid[i][0];
        }
        for(int j = 0; j < n; j++) {
            if(j == 0) dp[0][j] = grid[0][j];
            else dp[0][j] = dp[0][j-1] + grid[0][j];
        }
        for(int i = 1; i < m; i++) {
            for(int j = 1; j < n; j++) {
                dp[i][j] = min(dp[i-1][j], dp[i][j-1]) + grid[i][j];
            }
        }
        return dp[m-1][n-1];
    }
};